nix/nixpkgs.nix

```diff
@@ -2,18 +2,25 @@
 # SPDX-License-Identifier: Apache-2.0
 {
   lib,
+  self,
   inputs,
   ...
 }: {
+  # TODO include the cross-compilation for the relevant targets
+  # The cross-compilation should only be done if the target is a xcompile target, not a local target
+  # so a -from-x86
   perSystem = {system, ...}: {
-    # customise pkgs
+    # create a custom instance of nixpkgs with all our overlays and custom config
+    # and make it available to all perSystem functions
     _module.args.pkgs = import inputs.nixpkgs {
-      inherit system inputs;
+      inherit system;
+      overlays = import ../overlays;
       config = {
         allowUnfree = true;
       };
+      specialArgs = {
```

Comment:
Interesting. Didn't know that nixpkgs also has specialArgs. I only now this feature from NixOS.

Reply:
Mhm. Where is this defined? I cannot find this feature in nixpkgs.